roles/kubernetes/kubeadm/defaults/main.yml

```diff
@@ -4,9 +4,6 @@
 discovery_timeout: 60s
 kubeadm_join_timeout: 120s
 
-# Optionally remove kube_proxy installed by kubeadm
-kube_proxy_remove: false
```

Comment:
Why are you removing this var? It is referenced in several roles and needs a default value. kube-proxy and coredns are not skipped during kubeadm upgrade, so it always needs to be checked for removal

Reply:
This variable is already defined in kubespray-default (https://github.com/kubernetes-sigs/kubespray/blob/3cefd60c37cc2fc6d7ec13f10ed97e682b9a5dec/roles/kubespray-defaults/defaults/main.yaml#L28) and I believe that kubespray is moving most of default variable there.

Reply:
@mattymo Does this answer seem sufficient to you? If It is not I will be happy to quickly fix any problem on this PR...

Reply:
FYI, I just tested this again and the default value in other roles supersed those in kubespray-default. As the one in kubespray-default is slightly different It can cause issues in some environments (kube-router with service proxy)...